### DISCUSSION TITLE
Accessing $route params during data fetching

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/nuxt#18145] [Questions] [answered]
I have dynamic links to list of articles defined like that: 

`<NuxtLink :to="{ name: 'detail', params: { id: article.id } }">{{ article.title }}</NuxtLink>`

The data are flowing through, I can see corresponding values in `detail.vue` in `template `section using `{{ this.$route.params.id }}`.

But I want to make a dynamic API call to fetch article data. And I cannot figure out, how to access current `id `value in `script `section of `detail.vue`? this is undefined here, $route is undefined either. I feel like the answer must be pretty simple, but was unable to get it yet.

**Comment by viii47** [comment, marked as answer]
Hey @AloisSeckar,

when you want to access the route params, you need to reference the `useRoute()` function to have access to it.

Example:
```js

    [...]
    async setup() {
        const route = useRoute();
        console.log(route.params.id);
    }

```
